src/renderer/services/neo.js

```diff
@@ -383,187 +384,151 @@ export default {
+                decimals: fetchedBalance.assetId === NEO_ASSET_ID ? 0 : 8,
+                isUserAsset: true,
               };
-              if (restrictToSymbol && holdingBalance.symbol !== restrictToSymbol) {
-                return;
-              }
-              if (holdingBalance.symbol === 'NEO') {
-                promises.push(api.getMaxClaimAmountFrom({
-                  net: currentNetwork.net,
-                  url: currentNetwork.rpc,
-                  address: currentWallet.address,
-                  privateKey: currentWallet.privateKey,
-                }, api.neoscan)
-                  .then((res) => {
-                    holdingBalance.availableToClaim = toBigNumber(res);
-                  })
-                  .catch((e) => {
-                    alerts.networkException(e);
-                  }));
-              }
-
-              promises.push(dex.fetchContractBalance(holdingBalance.asset)
-                .then((res) => {
-                  holdingBalance.contractBalance = toBigNumber(res);
-                  holdingBalance.totalBalance = toBigNumber(holdingBalance.balance)
-                    .plus(holdingBalance.contractBalance).plus(holdingBalance.openOrdersBalance);
-                })
-                .catch((e) => {
-                  alerts.networkException(e);
-                }));
-              promises.push(dex.fetchOpenOrderBalance(holdingBalance.asset)
-                .then((res) => {
-                  holdingBalance.openOrdersBalance = toBigNumber(res);
-                  holdingBalance.totalBalance = toBigNumber(holdingBalance.balance)
-                    .plus(holdingBalance.contractBalance).plus(holdingBalance.openOrdersBalance);
-                })
-                .catch((e) => {
-                  alerts.networkException(e);
-                }));
-              holdings.push(holdingBalance);
+              _.set(holdingsToQueryBalance, systemAssetHolding.assetId, systemAssetHolding);
             });
 
-            tokens.getAllAsArray().forEach((nep5) => {
-              if (nep5.network !== currentNetwork.net) {
-                return;
-              }
+            const assetToHolding = (asset, isUserAsset) => {
+              const assetId = asset.assetId.replace('0x', '');
+              return {
+                assetId,
+                balance: new BigNumber(0),
+                symbol: asset.symbol,
+                name: asset.name,
+                isNep5: assetId.length === 40,
+                contractBalance: new BigNumber(0),
+                totalBalance: new BigNumber(0),
+                canPull: asset.canPull,
+                isUserAsset,
+              };
+            };
 
-              const inMemory = _.get(store.state.nep5Balances, nep5.assetId);
 
-              if (restrictToSymbol && nep5.symbol !== restrictToSymbol) {
-                return;
-              }
+            if (!lastVerifiedTokenBalances
```

Comment:
The needsRefresh flag is no longer being looked at. Isn't it still being set?

Reply:
For example:
[dex.js line 862](https://github.com/aphtoken/aph-wallet-gui/pull/132/files#diff-54fcb2cbbfe08304e72ac626427a1d01R862)
[dex.js line 904](https://github.com/aphtoken/aph-wallet-gui/pull/132/files#diff-54fcb2cbbfe08304e72ac626427a1d01R904)